include/platform/mir/graphics/egl_extensions.h

```diff
@@ -216,44 +220,83 @@ namespace graphics
 {
 struct EGLExtensions
 {
+    template<typename Ext>
+    struct LazyDisplayExtensions
+    {
+        auto operator()(EGLDisplay dpy) const -> Ext const&
+        {
+            if (!has_initialized)
+            {
+                std::lock_guard<std::mutex> lock{mutex};
+                if (!has_initialized)
+                {
+                    cached_display = dpy;
+                    cached_ext.emplace(dpy);
+                    has_initialized = true;
+                }
+            }
+
+            if (dpy != cached_display)
+            {
+                BOOST_THROW_EXCEPTION(std::logic_error("Multiple EGL displays used with the same extension object"));
+            }
+
+            return cached_ext.value();
```

Comment:
I *think* this is safe, but I'd appreciate @AlanGriffiths' review of this.

Reply:
Btw according to this totally reliable and guaranteed-to-be-correct [StackOverflow thread](https://stackoverflow.com/questions/40320254/reordering-atomic-operations-in-c) this sort of thing is safe

Reply:
Yes, the default on `atomic<>` is global sequential consistency. Hard thinking is only required of we loosen that requirement.